R/pkg/R/group.R

```diff
@@ -129,6 +129,48 @@ methods <- c("avg", "max", "mean", "min", "sum")
 # These are not exposed on GroupedData: "kurtosis", "skewness", "stddev", "stddev_samp", "stddev_pop",
 # "variance", "var_samp", "var_pop"
 
+#' Pivot a column of the GroupedData and perform the specified aggregation.
+#'
+#' Pivot a column of the GroupedData and perform the specified aggregation.
+#' There are two versions of pivot function: one that requires the caller to specify the list
+#' of distinct values to pivot on, and one that does not. The latter is more concise but less
+#' efficient, because Spark needs to first compute the list of distinct values internally.
+#'
+#' @param x a GroupedData object
+#' @param colname A column name
+#' @param values A value or a list/vector of distinct values for the output columns.
+#' @return GroupedData object
+#' @rdname pivot
+#' @name pivot
+#' @export
+#' @examples
+#' \dontrun{
+#' df <- createDataFrame(data.frame(
+#'     earnings = c(10000, 10000, 11000, 15000, 12000, 20000, 21000, 22000),
+#'     course = c("R", "Python", "R", "Python", "R", "Python", "R", "Python"),
+#'     year = c(2013, 2013, 2014, 2014, 2015, 2015, 2016, 2016)
+#' ))
+#' collect(sum(pivot(groupBy(df, "year"), "course"), "earnings"))
+#' collect(sum(pivot(groupBy(df, "year"), "course", "R"), "earnings"))
+#' collect(sum(pivot(groupBy(df, "year"), "course", c("Python", "R")), "earnings"))
+#' collect(sum(pivot(groupBy(df, "year"), "course", list("Python", "R")), "earnings"))
```

Comment:
nit: I think maybe we should vary the examples a bit? Perhaps to avoid promoting collect() if possible, either leave it at `pivot(....)` or set to a variable `group_sum <- sum(pivot(...))`